react/MenuItem.tsx

```diff
@@ -63,20 +66,31 @@ const submenuReducer: Reducer<SubmenuState, SubmenuAction> = (
         ...state,
         isActive: false,
       }
+    case 'DISABLE_ON_MOUNT_BEHAVIOR_FLAG':
+      return {
+        ...state,
+        onMountBehavior: 'closed',
+      }
     default:
       return state
   }
 }
 
 const MenuItem: StorefrontFunctionComponent<MenuItemSchema> = ({
   children,
+  onMountBehavior = 'closed',
   ...props
 }) => {
+
   const { experimentalOptimizeRendering } = useContext(MenuContext)
-  const [{ isActive, hasBeenActive }, dispatch] = useReducer(
-    submenuReducer,
-    submenuInitialState
-  )
+  const [
+    { isActive, hasBeenActive, onMountBehavior: onMountBehaviorFlag },
+    dispatch,
+  ] = useReducer(submenuReducer, {
+    hasBeenActive: onMountBehavior === 'open' ?? false,
+    isActive: onMountBehavior === 'open' ?? false,
```

Comment:
You don't need this `?? false`, right? The `===` will result in a boolean anyway

```suggestion
    hasBeenActive: onMountBehavior === 'open',
    isActive: onMountBehavior === 'open',
```